src/TTGO_T-Beam_LoRa_APRS.ino

```diff
@@ -480,27 +486,34 @@ String prepareCallsign(const String& callsign){
 }
 
 #if defined(ENABLE_TNC_SELF_TELEMETRY) && defined(KISS_PROTOCOL)
-void sendTelemetryFrame() {
-  #ifdef T_BEAM_V1_0
-    uint8_t b_volt = (axp.getBattVoltage() - 3000) / 5.1;
-    uint8_t b_in_c = (axp.getBattChargeCurrent()) / 10;
-    uint8_t b_out_c = (axp.getBattDischargeCurrent()) / 10;
-    uint8_t ac_volt = (axp.getVbusVoltage() - 3000) / 28;
-    uint8_t ac_c = (axp.getVbusCurrent()) / 10;
-
-    String telemetryParamsNames = String(":") + Tcall + ":PARM.B Volt,B In,B Out,AC V,AC C";
-    String telemetryUnitNames = String(":") + Tcall + ":UNIT.mV,mA,mA,mV,mA";
-    String telemetryEquations = String(":") + Tcall + ":EQNS.0,5.1,3000,0,10,0,0,10,0,0,28,3000,0,10,0";
-    String telemetryData = String("T#MIC") + String(b_volt) + ","+ String(b_in_c) + ","+ String(b_out_c) + ","+ String(ac_volt) + ","+ String(ac_c) + ",00000000";
-    String telemetryBase = "";
-    telemetryBase += Tcall + ">APLO01" + ":";
-    sendToTNC(telemetryBase + telemetryParamsNames);
-    sendToTNC(telemetryBase + telemetryUnitNames);
-    sendToTNC(telemetryBase + telemetryEquations);
-    sendToTNC(telemetryBase + telemetryData);
+  void sendTelemetryFrame() {
+    if(enable_tel == true){
+      #ifdef T_BEAM_V1_0
+        uint8_t b_volt = (axp.getBattVoltage() - 3000) / 5.1;
+        uint8_t b_in_c = (axp.getBattChargeCurrent()) / 10;
+        uint8_t b_out_c = (axp.getBattDischargeCurrent()) / 10;
+        uint8_t ac_volt = (axp.getVbusVoltage() - 3000) / 28;
+        uint8_t ac_c = (axp.getVbusCurrent()) / 10;
+        // Pad telemetry message address to 9 characters
+        char Tcall_message_char[9];
+        sprintf_P(Tcall_message_char, "%-9s", Tcall);
+        String Tcall_message = String(Tcall_message_char);
+
+        String telemetryParamsNames = String(":") + Tcall_message + ":PARM.B Volt,B In,B Out,AC V,AC C";
+        String telemetryUnitNames = String(":") + Tcall_message + ":UNIT.mV,mA,mA,mV,mA";
+        String telemetryEquations = String(":") + Tcall_message + ":EQNS.0,5.1,3000,0,10,0,0,10,0,0,28,3000,0,10,0";
+        String telemetryData = String("T#") + String(b_volt) + ","+ String(b_in_c) + ","+ String(b_out_c) + ","+ String(ac_volt) + ","+ String(ac_c) + ",00000000";
```

Comment:
This is looking good, except that the [telemetry sequence number](https://github.com/PhirePhly/aprs_notes/blob/master/telemetry_format.md) (`T#---`) is not being included at all.

Reply:
This is because I tore out `MIC` to make it work with aprs.fi. I plan to add a www toggle between MIC and sequence number, once I figure out how to store the sequence number.